docs/the-absolute-state-of-kde-software.org

```diff
@@ -243,8 +243,153 @@ And if you want a positive example; take =synaptic= or =octopi=.  Those programs
+
+**** Lesson 15:
+
+Don't rewrite it for the sake of rewriting.
+
+** Browsers
+
+This is perhaps a needlessly painful point, and I have no idea why this happened the way it happened and didn't happen any other way.
+
+We have many browser front ends, one back end, and mostly no agreement on why this terrible state of affairs erupted on the scene.  Probably because the Web is a hodgepodge of barely standard code.  Perhaps because the early web had decided to eschew the Unix approach for things and directly embed a JavaScript interpreter into the browser and couple it tightly to the layout engine.
+
+That is beside the point.  Currently, Falkon is the closest thing to an official KDE browser.  It has a plugin system, but no clear way of writing one, and even simple things as integrating KeePassXC into it seem to be problematic.  It is untenable, and while I would agree that writing a browser from the ground up would be too much hassle, I don't want people to do that.  Write a version of QuteBrowser that doesn't rely on modal editing[fn:4].  It's just that even Falkon to a large extent feels out of place on KDE.  It doesn't feel the same way /e.g./  Elisa or KTorrent feel, and isn't certainly nearly as polished as Kate or Dolphin.
+
+I wish we had a revival of konqueror, because the idea of a single purpose application that can render both file systems and browse the web is ideal.  The main issue is that I now believe that it would be taken to mean to write a file manager in Electron.
+
+**** Lesson 16:
+
+I don't even know what to say.
+
+** Dolphin
+
+I want to round off this section with a discussion of an example of almost all of the lessons applied.  Dolphin is a remarkable application that brings many of the things that make KDE -- KDE, to the forefront and can be considered by many to be KDEs killer app.
+
+Dolphin is a file manager that fundamentally understands that you want to manage files.  It's not quite the swiss army knife that konqueror used to be but that is for the better.  Konqueror would have been a mediocre browser and trying to make it passable would have pulled resources away from making it a good file manager.  And there's a lot to Dolphin.
+
+Want to have a UI that resembles YouTube with local videos, you can do it, just crank up the preview sizes.  Want to make things as clear as a table, sure, there's that view too.  OK, how about features that other file managers don't have? Well, Dolphin and Finder are to my knowledge the only two that also preserve the settings on a per-directory basis.  Certainly the GTK file managers could do the same, but we don't assign credit based on hypotheticals.
```

Comment:
I think you can do much more than preview sizes, but this is rarely done and it's being done in different software.

Let's say I'm walking around and filming city an hour. How about storing section previews where the time or color palette changed a lot? Maybe I'm filming / visiting some nice mall I want to see. TL;DR, a lot more can be done about the preview mechanism.